.github/workflows/unit-tests.yaml

```diff
@@ -0,0 +1,24 @@
+name: unit-tests
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '**.go'
+      - Makefile
+      - '**.tpl'
+      - go.mod
+      - go.sum
+
+jobs:
+  unit-tests:
+    name: make test
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.15'
```

Comment:
technically, our go.mod files specify 1.14, but meh, not sure it's a huge deal here.